contracts/MetadataViews.cdc

```diff
@@ -238,4 +239,61 @@ pub contract MetadataViews {
             self.url=url
         }
     }
+
+
+    // A view to expose the information needed store and retrieve an NFT
+    //
+    // This can be used by applications to setup a NFT collection with proper storage and public capabilities.
+    pub struct NFTCollectionData {
+        // Path in storage where this NFT is recommended to be stored.
+        pub let storagePath: StoragePath
+
+        // Public path which must be linked to expose public capabilities of this NFT
+        // including standard NFT interfaces and metadataviews interfaces
+        pub let publicPath: PublicPath
+
+        // Private path which should be linked to expose the provider
+        // capability to withdraw NFTs from the collection holding NFTs
+        pub let providerPath: PrivatePath
+
+        // Public collection type that is expected to be linked at the aforementioned public path
+        // and provides sufficient access to standard functions (i.e. deposit + getIDs + borrowNFT)
+        // This is normally a restricted type with a single interface.
+        pub let publicCollection: Type
+
+        // Type that should be linked at the aforementioned public path. This is normally a
+        // restricted type with many interfaces. Notably the `NFT.CollectionPublic` and
+        // `NFT.Receiver` are required.
+        pub let publicLinkedType: Type
```

Comment:
I'm still a little bit confused on the difference between `publicCollection` and `publicLinkedType` and why we would need both of them. Can you provide more clarification?

Reply:
The reason for both is for backwards compatibility with existing collections.

Many collections have not been linking NFT.CP because it was not enforced in the past, and only their specific NFT's CollectionPublic exposed by accounts that hold their collections.

Having this separate publicCollection which is guaranteed to work for the `read` use-case will allow DAPPs (like marketplaces) to read from collections in a standard way. And the `publicLinkedType` will serve the purpose of supporting setup or re-linking the public path in the proper way with standard collections.

Reply:
ok, that makes sense. Can you add a bit of that clarification in the comments as well?